### PR TITLE
Port indent-to

### PR DESCRIPTION
**Pull request by krobelus**
Not sure whether we should use EmacsUint or EmacsInt for a column number.

### CODE REVIEW

File: rust_src/src/buffers.rs [hunk 1]
```diff
@@ -177,6 +177,10 @@ impl LispBufferRef {
         self.case_fold_search_
     }
 
+    pub fn tab_width(self) -> LispObject {
```
Comment (shaleh) [inline rust_src/src/buffers.rs]: Just use `tab_width_`. No need for a method.

File: rust_src/src/indent.rs [hunk 2]
```diff
@@ -134,4 +141,46 @@ pub extern "C" fn invalidate_current_column() {
     unsafe { last_known_column_point = 0 };
 }
 
+/// Indent from point with tabs and spaces until COLUMN is
+/// reached. Optional second argument MINIMUM says always do at least
+/// MINIMUM spaces even if that goes past COLUMN; by default, MINIMUM
+/// is zero.
+///
+/// The return value is the column where the insertion ends.
+#[lisp_fn(min = "1")]
```
Comment (brotzeit) [inline rust_src/src/indent.rs]: The intspec is missing `"NIndent to column: "`

Comment (shaleh) [inline rust_src/src/indent.rs]: You missed the `intspec`. rust_src/rustize-remacs is a helper script you can call with `M-|` to automatically port the DEFUN comments and make a new lisp_fn for you. Give it a try.

Comment (krobelus) [reply rust_src/src/indent.rs]: awesome :+1:

File: rust_src/src/indent.rs [hunk 3]
```diff
@@ -134,4 +141,41 @@ pub extern "C" fn invalidate_current_column() {
...
+/// Indent from point with tabs and spaces until COLUMN is reached.
+/// Optional second argument MINIMUM says always do at least MINIMUM
+/// spaces even if that goes past COLUMN; by default, MINIMUM is zero.
+///
+/// The return value is the column where the insertion ends.
+#[lisp_fn(min = "1", intspec = "NIndent to column: ")]
+pub fn indent_to(column: EmacsInt, minimum: Option<EmacsInt>) -> EmacsInt {
+    let buffer = ThreadState::current_buffer_unchecked();
+    let tab_width = unsafe { sanitize_tab_width(buffer.tab_width_.force_fixnum()) } as EmacsInt;
+    let arg_minimum = minimum.unwrap_or(0);
```
Comment (shaleh) [inline rust_src/src/indent.rs]: :+1:

File: src/indent.c [hunk 4]
```diff
@@ -779,54 +779,6 @@ string_display_width (Lisp_Object string, Lisp_Object beg, Lisp_Object end)
...
-even if that goes past COLUMN; by default, MINIMUM is zero.
-
-The return value is the column where the insertion ends.  */)
-  (Lisp_Object column, Lisp_Object minimum)
-{
-  EMACS_INT mincol;
-  register ptrdiff_t fromcol;
-  int tab_width = SANE_TAB_WIDTH (current_buffer);
-
-  CHECK_NUMBER (column);
```
Comment (shaleh) [inline src/indent.c]: `CHECK_NUMBER` validates that column is an integer. But it does not check for positive or negative. So `EmacsInt` is correct. If instead either `NATNUMP` or `CHECK_NATNUM` is used then `EmacsUint` is appropriate.

**Pull request merged**